apps/sms/views/shared/test/unit/system_message_handler_test.js

```diff
@@ -0,0 +1,778 @@
+                  closeOnClick: false
+                }
+              );
+            }
+          ).then(done, done);
+        });
+
+        test('message has iccId and several SIMs are presented',
+        function(done) {
+          SystemMessageHandler.onSmsReceivedSystemMessage(smsMessage).then(
+            () => {
+              sinon.assert.calledWith(
+                NotificationHelper.send,
+                {
+                  id: 'dsds-notification-title-with-sim',
+                  args: { sim: 'SIM 2', sender: smsMessage.sender }
+                },
+                {
+                  icon: 'iconURI',
+                  bodyL10n: { raw: smsMessage.body },
+                  tag: 'threadId:' + smsMessage.threadId,
+                  data: { id: smsMessage.id, threadId: smsMessage.threadId },
+                  closeOnClick: false
+                }
+              );
+            }
+          ).then(done, done);
+        });
+
+        test('message from contact with iccId and several SIMs are presented',
+        function(done) {
+          Contacts.findByAddress.withArgs(smsMessage.sender).returns(
+            Promise.resolve(MockContact.list())
+          );
+
+          SystemMessageHandler.onSmsReceivedSystemMessage(smsMessage).then(
+            () => {
+              sinon.assert.calledWith(
+                NotificationHelper.send,
+                {
+                  id: 'dsds-notification-title-with-sim',
+                  args: { sim: 'SIM 2', sender: 'Pepito O\'Hare' }
+                },
+                {
+                  icon: 'iconURI',
+                  bodyL10n: { raw: smsMessage.body },
+                  tag: 'threadId:' + smsMessage.threadId,
+                  data: { id: smsMessage.id, threadId: smsMessage.threadId },
+                  closeOnClick: false
+                }
+              );
+            }
+          ).then(done, done);
+        });
+      });
+    });
+  });
+
+  suite('notification system message', function() {
+    var appStub, smsMessage;
+
+    setup(function() {
+      smsMessage = MockMessages.sms();
+      appStub = sinon.stub({ launch() {} });
+
+      this.sinon.stub(navigator.mozApps, 'getSelf').returns(
+        Promise.resolve(appStub)
+      );
+
+      this.sinon.stub(Navigation, 'isCurrentPanel').returns(false);
+      this.sinon.stub(Navigation, 'toPanel').returns(Promise.resolve());
+
+      this.sinon.stub(MessageManager, 'getMessage').withArgs(
+        smsMessage.id
+      ).returns(smsMessage);
+    });
+
+    test('nothing is done when notification is removed', function(done) {
+      SystemMessageHandler.onNotificationSystemMessage({
+        // When notification is removed "clicked" property is "false".
+        clicked: false,
+        data: { id: smsMessage.id, threadId: smsMessage.threadId }
+      }).then(() => {
+        sinon.assert.notCalled(appStub.launch);
+        sinon.assert.notCalled(Navigation.toPanel);
```

Comment:
very, very optional nit: Since you would like to insert a newline in front of the `Navigation.toPanel` assertion, maybe you would like to sync the convention here?